torch/ao/quantization/_quantize_pt2e.py

```diff
@@ -0,0 +1,205 @@
+import torch
+from torch.fx import GraphModule
+import torch._dynamo as torchdynamo
+from torch.nn.utils.fusion import fuse_conv_bn_weights
+
+from torch.ao.quantization.fx.prepare import (
+    is_activation_post_process_node,
+)
+from .qconfig_mapping import QConfigMapping
+from .backend_config import BackendConfig
+from .fx import prepare
+from .quantize_fx import _convert_to_reference_decomposed_fx
+
+from typing import Tuple, Any, Dict
+from collections import OrderedDict
+import operator
+
+# TODO: longer term, don't need to retrace or parse the string
+# we should have node.meta["nn_module_stack"] that store the dict
+def _infer_nn_stack_trace_and_append_on_mode(m, gm, args_as_list):
+    trace_func, guards = torchdynamo.export(
+        m,
+        args_as_list,
+        aten_graph=True,
+        tracing_mode="real"
+    )
+    reset_metadata = {}
+    for node in trace_func.graph.nodes:
+        nn_module_stack = {}
+        if (stack_trace := node.meta.get("stack_trace")) is not None:
+            for line in stack_trace.split("\n"):
+                if line.startswith("Module stack:"):
+                    mod_trace = eval(line.replace("Module stack:", ""))  # pyre-ignore
+                    nn_module_stack = {"nn_module_stack": mod_trace}
+        reset_metadata[node.name] = nn_module_stack
+
+    for n in gm.graph.nodes:
+        if (meta := reset_metadata.get(n.name)):
+            n.meta.update(meta)
+
+# TODO: longer term, this should happen in the dynamo stack as well
+def _get_renamed_nn_module_stack(nn_module_stack):
+    # initialize with top level parent scope
+    nn_module_stack_renamed = OrderedDict([("", None)])
+    if nn_module_stack:
+        # Rename module_key, e.g. "self_layer1_1__conv1" to "self.layer1.1._conv1", for easier downstream parsing
```

Comment:
How would this distinguish `self.module_1 = ...`. You would replace this with self.module.1, no? But that woudl be wrong. Basically _ can be part of the attribute name.

Reply:
this is fine, we will look at what is the previous key and use that to figure out the current module path, in the example you gave, it will be something like the following:
{"self_module_1": ..., "self_module_1_abc": ...}
when we parse the first key, we'll get "self.module_1", when we parse the second key, we'll have "self.module_1.abc"